compiler/rustc_metadata/src/rmeta/decoder.rs

```diff
@@ -1883,6 +1898,10 @@ impl CrateMetadata {
         self.dependencies.push(cnum);
     }
 
+    pub(crate) fn target_modifiers(&self) -> &TargetModifiers {
+        &self.target_modifiers
```

Comment:
This function seems to be called only once. Maybe do the decoding inside this function instead of storing it in `CrateMetadata`?